pymeasure/instruments/hp/hp8657b.py

```diff
@@ -0,0 +1,188 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2022 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+
+import logging
+from pymeasure.instruments import Instrument
+from pymeasure.instruments.validators import strict_discrete_set, strict_range
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+class HP8657B(Instrument):
+    """ Represents the Hewlett Packard 8657B signal generator
+    and provides a high-level interface for interacting
+    with the instrument.
+    """
+
+    def __init__(self, adapter, **kwargs):
+        super().__init__(
+            adapter,
+            "Hewlett-Packard HP8657B",
+            includeSCPI=False,
+            send_emd=True,
+            **kwargs,
+        )
+
+    def check_errors(self):
+        """
+        Method to read the error status register
+
+        as the 8657B does not support any readout of values, this will return 0 and log a warning
+
+        """
+        log.warning("HP8657B Does not support error status readout")
+
+    def clear(self):
+        """
+        Reset the instrument to power-on default settings
+
+        """
+        self.adapter.connection.clear()
+
+    id = "HP 8657B Signal generator"  #: Manual ID entry, as the instrument does not talk with us
+
+    # List of modulation sources
+    modulation_source = {"External": 1,
+                         "Int_400Hz": 2,
+                         "Int_1000Hz": 3,
+                         "OFF": 4,
+                         "DC_FM": 5}
+
+    am_depth = Instrument.setting(
+        "AM %2.1f PC",
+        """
+        sets the modulation depth for AM,
+        usable range 0-99.9%
```

Comment:
The property docstrings should follow the normal docstring style (according to the not yet merged documentation overhaul). Therefore the first sentence is an imperative (and starts with a capital letter):
```suggestion
        Set the modulation depth for AM,
        usable range 0-99.9%
```

Additionally, I'd add the abbreviation definitions of "AM" and "FM" to the device docstring, just in case. I guess they mean amplitude and frequency modulation.

Reply:
added a shot glossary with https://github.com/pymeasure/pymeasure/pull/732/commits/6d61a9ae93fb0335ed54913de60c07202716f59d